code/lib/cli/src/js-package-manager/JsPackageManager.ts

```diff
@@ -76,6 +76,49 @@ export abstract class JsPackageManager {
     this.cwd = options?.cwd || process.cwd();
   }
 
+  /** Detect whether Storybook gets initialized in a monorepository/workspace environment
+   * The cwd doesn't have to be the root of the monorepo, it can be a subdirectory
+   * @returns true, if Storybook is initialized inside a monorepository/workspace
+   */
+  public isStorybookInMonorepo() {
+    let cwd = process.cwd();
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
```

Comment:
you could use `find-up` to achieve the same thing, I believe. It's a dependency of the CLI already

Reply:
I am not sure how `find-up` could help here. It is not just about finding a file but also about evaluating its content. If the `<package.json>.workspace` is not set, I still want to continue trailing up.